### PR TITLE
Lavaland loot changes

### PR DESCRIPTION
**Pull request by Nazsgull**
## What Does This PR Do

Removes the guardian spawner from the tendril loot pool and adds it to the Legion loot pool.

Internally, the chests Legion may drop when killed are now a different object from the tendril ones. They have the same appearance but the name, description, and set of items they can drop are different.

## Why It's Good For The Game
Miners were able to get 12TC worth of syndicate equipment in the first minute of each shift for a little challenge.
By moving it to the Legion loot pool, only one miner can get the item on each shift, and said item (guardian spawner) cannot be acquired at the beginning of the round.

Internally speaking it is a good thing to be able to separate the different loot pools from Lavaland, for ease of tweaks.

## Changelog
:cl:
add: Legion chests, which will drop the usual loot, including guardian tarot cards.
del: guardian tarot cards will no longer drop from tendril chests.
tweak: Legion and tendril loot pool is now different.
/:cl:

### CODE REVIEW

**Comment by Cocacolagua** [comment]
I agree with the different loot table for the legion chest, would be great to expand it in another PR so the chest has it’s own "theme", great PR anyway

**Comment by shmabuu** [comment]
Yep. Please do this. Thank you have a nice day.

**Comment by TheMaveran** [comment]
Worth noting that this doesn’t make it slightly rarer, it makes it extremely rare. Like 1 in a 1000 rounds, as legions chance of dropping a chest is low, and then it being tarot cards is even lower. Assuming someone even kills legion.

**Comment by Pennwick** [comment]
I'm gonna echo TheMaveran's concern there. You now have a 1 in 27 chance of getting it from a legion chest. I haven't played a ton of miner but I don't recall even ever seeing the legion downed. 

An alternative would just be to mess with the weights of items in the chests.

**Comment by Mitchs98** [comment]
A better option would be to just make guardian cards rarer to find in the chests, rather than moving them to something that'd basically make it nigh impossible to ever get it(or at-least extremely rare to the point it probably won't happen), as I mentioned before rather than going the much more heavy handed route as is here.

**Review by Kyep — APPROVED** [review]

**Comment by Spacemanspark** [comment]
I'm perfectly fine with this. If you want something that's worth 12tc, you gotta work for it.

File: code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm [hunk 1]
```diff
@@ -141,3 +141,80 @@ Difficulty: Medium
...
+	name = "Legion chest"
+	desc = "You feel the pressure of many thousand souls looking at you angrily."
+
+/obj/structure/closet/crate/necropolis/legion/New(add_loot = TRUE)
+	..()
+
+	if(!add_loot)
+		return
+
+	var/loot = rand(1, 26)
```
Comment (Qwertytoforty) [inline code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm:155]: Why is this rand 1, 26 when there is 27 items in this chest as options.

**Comment by Cocacolagua** [comment]
On second thought, changes are required yeah.

Each legion skull has a 5% of dropping the chest, and there are eight small skulls, so it’s a 40% for the chest to drop, and then a 1/27 to get the item.

I would suggest removing most of the garbage items from the chest, like cult bedsheets and the pick axe, so you don’t get absolutely scammed. (And in the future giving the legion their own exclusive items)

File: code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm (hunk 1 above)
Comment (Nazsgull) [reply code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm]: Yeah that is a whoopsie I made. Fixed in bbab42d43a248a77477da047c878e483aa260114

**Comment by Pixelzilla** [comment]
>>I'm perfectly fine with this. If you want something that's worth 12tc, you gotta work for it.

So actively fighting and hunting down tendrils, avoiding megafauna and other deadly elements from lavaland and then managing to get rng luck of getting this item instead of one of the numerous other more common trash items isn't work? Compared to rolling RNG to get traitor and spending TC that you didnt have to earn? It isn't even the same item and it isn't like getting it makes you an antag either, you're part of the crew and you cant just abuse the heck out of having a guardian.

Locking guardians meant to help their users behind something like Legion is pretty lame when the odds of getting it is fairly low in the first place and it probably was originally intended for miners to use to fight/survive the megafauna in the first place. You have to get lucky with there being a lot of tendrils, hope you manage to get one before another miner and still cross your fingers and hope you manage to get it and that the guardian you get is good.

**Comment by Spacemanspark** [comment]
It's painfully easy to destroy most tendrils, and people cheese lavaland enough as is.

**Comment by ppi13** [comment]
with the GPS being removed from tendrils as is, this is less of an issue.
However, instead of copypasting the drops from tendril chest, why not have the guardian be one outta two possible drops from killing the last body, so you either get staff or guardian, this way legion gives something actually good, it's does not become a lolrng, thing either.

**Comment by Pixelzilla** [comment]
>>It's painfully easy to destroy most tendrils, and people cheese lavaland enough as is.

What about all the other points that i raised? If we're going to weigh the "difficulty" of doing something then even the painfully easy to destroy tendrils are still more difficult and require more action than just rolling traitor via rng, opening your uplink and buying a holopara.

**Comment by Spacemanspark** [comment]
Traitors are intended to have holoparasites. That's not a fair comparison. 

For anything else, there should be a genuine challenge to obtaining one. There's already enough ridiculous loot miners can get as is.

**Review by farie82 — APPROVED** [review]
LGTM. Bit odd to dupe the whole chests inventory but it will be easier to change it in this format.

**Comment by Pixelzilla** [comment]
>>Traitors are intended to have holoparasites. That's not a fair comparison.

For anything else, there should be a genuine challenge to obtaining one. There's already enough ridiculous loot miners can get as is.

You're right it absolutely isn't a fair comparison to compare a traitor who is an antag with a holoparasite to a miner who is probably not a traitor with a guardian. Which is part of my entire point. You're treating them like they are the same items in the same situations and will cause equal devastation which they cannot due to self-antagging rules.

Is it true that a traitor could also be a miner and find one? Yes. But that means that person lucked out with rng multiple times and in my opinion that means they deserve it and it isn't like it even happens all the time. Its rng. Its luck. That traitor had to risk their antag round to go out and hunt/find and pray to RNG gods that they actually get the cards. They deserve it.

No one argued against traitors having a holoparasite and no one said holoparasites should be mining loot. Holoparasites and Guardians are not the same thing. This is why there is a difference between them. This is why they are named differently and sport different sprites to define them as similarly inspired but different. 

In addition adding it to Legion is probably going to have more of a negative impact on other people than having it in tendril loot. Most of the time people do not release Legion. If this goes through it means we'll probably see more and more shifts where mining is out of action because of Legion being released and inevitably drawn closer to the outpost making it impossible for other miners who didn't want to fight Legion to actually do the jobs they sign up for.

**Comment by Spacemanspark** [comment]
>You're right it absolutely isn't a fair comparison to compare a traitor who is an antag with a holoparasite to a miner who is probably not a traitor with a guardian. Which is part of my entire point. You're treating them like they are the same items in the same situations and will cause equal devastation which they cannot due to self-antagging rules.

I'm not assuming they're going to be self antagging. That's a bit presumptuous of you.
The issue lies with validhunting and just being an overwhelming force to deal with. I've seen miners stacked with gear and a guardian just mow down blobs or anything that dares cross their sights. I've seen them in cult rounds where they decimate security singlehandedly, I've even seen them absolutely trash terrors before. Miners in general have easy access to ridiculous gear, guardian or not. _my_ point is that they need it to be toned down. The tendrils are unbelievably easy to take down with a kinetic accelerator and a small amount of distance most of the time. Actually, hell, I walked up to one as an ash walker with a spear the other day and ripped it apart easily.

>Is it true that a traitor could also be a miner and find one? Yes. But that means that person lucked out with rng multiple times and in my opinion that means they deserve it and it isn't like it even happens all the time. Its rng. Its luck. That traitor had to risk their antag round to go out and hunt/find and pray to RNG gods that they actually get the cards. They deserve it.

See my point above. Tendrils are easy to destroy. I don't find it an inherent risk at this point, outside of the newest of new miners. The only thing you're risking is wasting time.

>No one argued against traitors having a holoparasite and no one said holoparasites should be mining loot. Holoparasites and Guardians are not the same thing. This is why there is a difference between them. This is why they are named differently and sport different sprites to define them as similarly inspired but different.

This point doesn't make much sense to me. If I renamed and resprited a pulse rifle and put it into the list of loot, does that make it completely and utterly different? No, because code wise and gameplay wise they're pretty much the same thing.

>In addition adding it to Legion is probably going to have more of a negative impact on other people than having it in tendril loot. Most of the time people do not release Legion. If this goes through it means we'll probably see more and more shifts where mining is out of action because of Legion being released and inevitably drawn closer to the outpost making it impossible for other miners who didn't want to fight Legion to actually do the jobs they sign up for.

If there are miners throwing themselves at legion round after round to the point where it begins to hinder gameplay for other players, that falls onto the admin team to handle.
Alternatively, we could just remove it from the loot pool outright. I have no issues with either option.

**Comment by GentleManners** [comment]
Guardian/Holopara fall mostly into the more gun than useful category. I don't get why exactly that item is an issue. Also on that note, saying that mining is broken is a bit dishonest. Did miners wreck an antag at some point or another with their gear?Sure but that's robust players/validhunters. How many times did you see an officer mow down the cult or officers or scientist with all the gear/a mech suit or a botanist with super meme chems turning into an unstoppable and unstunable killing machine? Acting like mining is the most broken thing ever is odd, considering it's the only department where you actually risk your life to get your gear. Doy that logic do you want to remove robotics abilities to build durands/gygax or xray guns too because they mess with blobs too much

**Comment by Cocacolagua** [comment]
We... actually do want to remove/rework mechas and hate xray guns, yes. Mining, science and botany are all the biggest issue in the terms of this game’s balancing, and the reason mining is the most nerfed in these last PRs is that it’s the easiest one to nerf. Nerfing botany’s and science’s deathmixes, or RND in it’s entirety is... quite hard, given that you don’t know where to start from.

I’d say the biggest issue with mining is that a lot of it’s loot (not all, but most) falls under the category of: Useless for mining, good for fragging antags (which we want to discourage) and fragging as antag (which nobody likes to see). And this loot is obtained by playing a singleplayer game in a (supposedly) multiplayer roleplay game.

**Comment by GentleManners** [comment]
Yes you are right but we are going down the slippery slope of making mining
not worth it. It is single player yes but mining still has a lot of
randomness, risk and time investment needed for the good stuff so the items
should be a little bit more rewarding than for example a scientist walking
up to a rnd console ans pressing one button. Otherwise it will turn joke
only like you mentioned. For example if you get a pickaxe, one of the mods,
robes etc there is nothing of value there, you might as well throw it away

Cocacolagua <notifications@github.com> schrieb am Di., 22. Sep. 2020, 16:12:

**Comment by Spacemanspark** [comment]
If making mining 'worth it' means feeding them ridiculously overpowered gear, then it may be best to reconsider having lavaland.

**Comment by Cocacolagua** [comment]
In an ideal world, we would rework lavaland into being a mining-only experience, and the “PvE combat” can be moved into the gateway, which also needs a purpose.

But given that would take a lot of effort, we are stuck with the nerf-ening of mining items until they’re no longer overpowered at destroying whatever there is on-station.

The same happens with RnD and botany, given that re-working both or any of them would take a ludicrous amount of effort, people just don’t try to balance them to not being bullshit.

**Comment by GentleManners** [comment]
Mining has its balance issues but saying its ridiculously overpowered and needs 0 effort to get seems a bit harsh

**Comment by Spacemanspark** [comment]
Not really

**Comment by Kyep** [comment]
Not even a little bit harsh. Mining is most definitely OP.

**Comment by Alantodne** [comment]
The UI got in the way of my bag mutiple times, preventing me from using a rune and instead pulling a item out of my bag.

**Comment by AllStickyGithub** [comment]
Good step in the right direction.

**Comment by AffectedArc07** [comment]
Please deconflict and update

File: code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm [hunk 2]
```diff
@@ -141,3 +141,80 @@ Difficulty: Medium
 	gpstag = "Echoing Signal"
 	desc = "The message repeats."
 	invisibility = 100
+
+/obj/structure/closet/crate/necropolis/legion
+	name = "Legion chest"
```
Comment (dearmochi) [inline code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm:146]:
```suggestion
	name = "\improper Legion chest"
```

**Review by dearmochi — APPROVED** [review]

**Comment by AffectedArc07** [comment]
I asked for this to be deconflicted on March 30th

It is now July 18th


Feel free to reopen whenever

**Pull request closed without merging**